From 03443ac1550744e151525483f74ecb921bbf10f6 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:30 -0400
Subject: [ethernet] qlcnic: Fix PVID configuration on eSwitch port

Message-id: <1408626422-3941-107-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88663
O-Subject: [PATCH RHEL 7.1 106/138] qlcnic: Fix PVID configuration on eSwitch port.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id a78b6da89f52d03997619f5a78a5325bec865977

Clear older PVID before adding a newer PVID to the eSwicth port

Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ctx.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ctx.c
index a81ad50..c1e11f5 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ctx.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ctx.c
@@ -1354,6 +1354,7 @@ int qlcnic_config_switch_port(struct qlcnic_adapter *adapter,
    arg2 &= ~BIT_3;
   break;
  case QLCNIC_ADD_VLAN:
+   arg1 &= ~(0x0ffff << 16);
    arg1 |= (BIT_2 | BIT_5);
    arg1 |= (esw_cfg->vlan_id << 16);
    break;
-- 
1.7.1